diff --git a/core/src/main/scala/org/apache/spark/util/ConsolidatedListingManager.scala b/core/src/main/scala/org/apache/spark/util/ConsolidatedListingManager.scala
deleted file mode 100644
index d81d4adcdf9..00000000000
--- a/core/src/main/scala/org/apache/spark/util/ConsolidatedListingManager.scala
+++ /dev/null
@@ -1,320 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.spark.util
-
-import java.io.{FileNotFoundException, IOException}
-
-import scala.annotation.tailrec
-import scala.collection.mutable.IndexedSeqView
-import scala.util.{Failure, Success}
-
-import org.apache.hadoop.conf.Configuration
-import org.apache.hadoop.fs.{FileStatus, FileSystem, Path}
-
-import org.apache.spark.internal.Logging
-
-/**
- * Listing Manager to list leaf files using the statOffset. It's a stateful class, hold on to the
- * listing response which is used for future requests.
- */
-
-private[spark] class ConsolidatedListingManager(
-    hadoopConf: Configuration,
-    fileSystemProvider: (Path, Configuration) => FileSystem)
-    extends Logging {
-
-  // path which was lastRequested
-  private var lastRequested: Option[Path] = None
-
-  /**
-   * Manages the call to underneath fileSystem. Keep tracks of the response aka batch and current
-   * item in the batch which needs to be mapped to a partition.
-   */
-  private object BatchManager {
-    private var batchScore: Int = 1
-    private var currentBatch: Option[IndexedSeqView[FileStatus, Array[FileStatus]]] = None
-    private var currentIndex: Int = 0
-
-    private def updateBatch(
-        newBatch: Option[IndexedSeqView[FileStatus, Array[FileStatus]]]): Unit = {
-      currentBatch = newBatch
-      currentIndex = 0
-      batchScore = 1
-    }
-
-    def fetchNextPage(): Unit = {
-      // Use a guard clause to exit early if the resource is exhausted.
-      if (isResourceExhausted) return
-
-      currentBatch match {
-        case Some(page) if page.nonEmpty =>
-          val lastItem = page.last
-          val response = fetchFileStatusStartingFrom(lastItem.getPath)
-          // Remove duplicate first element from next page fetch
-          // Duplicate would be there as statOffset API includes request path as well.
-          // (given it is a file)
-          val nextPageContent =
-            if (response.headOption.map(_.getPath).contains(lastItem.getPath)) {
-              response.view.drop(1)
-            } else {
-              // Otherwise, create a view of the entire response.
-              response.view
-            }
-
-          updateBatch(Some(nextPageContent))
-
-        case Some(_) =>
-        // If the current page exists but is empty, there's nothing to do.
-
-        case None =>
-          // This case remains the same, as throwing an exception for an invalid
-          // state is the correct behavior.
-          throw new IllegalStateException(
-            "Cannot fetch next page when current page is not defined." +
-              " Control should not reach here.")
-      }
-    }
-
-    def fetchFreshBatch(path: Path): Unit = {
-      val resp = fetchFileStatusStartingFrom(path)
-      updateBatch(Some(resp.view))
-    }
-
-    def incrementBatchScore(): Unit = {
-      batchScore += 1;
-    }
-
-    def fetchBatchScore(): Int = {
-      batchScore
-    }
-
-    /**
-     * Determines if requested path can be served from the current response.
-     */
-    def canServeFromCurrentResponse(targetPath: Path): Boolean = {
-      isOrderedRequest(targetPath) && currentBatch.exists { responseArray =>
-        responseArray.nonEmpty && {
-          val lastPath = responseArray.last.getPath
-          lastPath.toString.compareTo(targetPath.toString) >= 0
-        }
-      }
-    }
-
-    def currentItem: Option[FileStatus] = {
-      currentBatch.flatMap(_.lift(currentIndex))
-    }
-
-    def incrementIndex(): Unit = {
-      currentIndex += 1
-    }
-
-    /**
-     * If current page fetched from gcs is empty i.e. we don't have any more file left.
-     */
-
-    def isResourceExhausted: Boolean = {
-      currentBatch.exists(_.isEmpty)
-    }
-
-    /**
-     * Makes a call to list API for a provided path.
-     */
-    private def fetchFileStatusStartingFrom(path: Path): Array[FileStatus] = {
-      val fs = fileSystemProvider(path, hadoopConf)
-      val listStatusMethod = ListingUtil.listMethodOptimized(fs)
-      listStatusMethod match {
-        case Success(method) =>
-          val result = ListingUtil.invokeMethodJavaReflection[Array[FileStatus]](method, fs, path)
-          result match {
-            case Success(res) =>
-              logDebug(
-                s"call to new list call successful. Got ${res.length} files for " +
-                  s"path: ${path}")
-              res
-            case Failure(res) =>
-              logWarning(
-                s"call to new list call failed for " +
-                  s"path: ${path}")
-              throw new IOException("issue while listing files", res)
-          }
-        case Failure(e) =>
-          throw new IOException("listStatusStartingFrom is not supported", e)
-      }
-    }
-  }
-
-  /*
-   * Requested path are in correct order if
-   * 1. they are lexicographically in correct order
-   * 2. not a sub-directory/partition of previously served path
-   *
-   * If A/ followed by A/B/, it is not in correct order
-   * Although they are in lexicographical order but because it's a sub directory of previous
-   * served path. We must have already listed all file in A/B. Current index in the page
-   * wouldn't be pointing to correct value to fulfill A/B.
-   * Hence a fresh request needs to be made for A/B.
-   */
-  private def isOrderedRequest(requested: Path): Boolean = {
-    lastRequested.forall { previousPath =>
-      val requestedDirStr = ListingUtil.toDir(requested).toString
-      val previousDirStr = ListingUtil.toDir(previousPath).toString
-
-      previousDirStr.compareTo(requestedDirStr) < 0 &&
-      !ListingUtil.isChildOf(previousPath, requested)
-    }
-  }
-
-  private def fetchNewBatchIfNeeded(path: Path): Unit = {
-    if (!BatchManager.canServeFromCurrentResponse(path)) {
-      // can't serve path from existing batch, fetch new
-      BatchManager.fetchFreshBatch(path)
-    } else {
-      // If request can be served from existing batch
-      // update the score for batch
-      BatchManager.incrementBatchScore()
-    }
-  }
-
-  private def updateServedPath(path: Path): Unit = {
-    lastRequested = Option.apply(path)
-  }
-
-  def listLeafFilesV2(path: Path): Array[FileStatus] = {
-    logInfo(s"Requesting listing leaf files for ${path}")
-    try {
-      val dirPath = ListingUtil.toDir(path)
-      fetchNewBatchIfNeeded(dirPath)
-      @tailrec
-      def collect(collected: List[FileStatus]): List[FileStatus] = {
-        BatchManager.currentItem match {
-          case Some(status) =>
-            val statusPath = status.getPath
-            if (ListingUtil.isChildOf(dirPath, statusPath)) {
-              // It's a child file, collect it and continue with the next item.
-              logDebug(s"Path ${statusPath} qualify as a leaf for ${dirPath} collect it.")
-              BatchManager.incrementIndex()
-              collect(status :: collected)
-            } else if (statusPath == path) {
-              // Path itself is a file. Collect it and stop.
-              BatchManager.incrementIndex()
-              logDebug(s"Path ${dirPath} is a file, not a directory. Collected it directly.")
-              status :: collected
-            } else {
-              val statusPath = status.getPath
-              if (statusPath.toString < dirPath.toString) {
-                // skip this status as it is pointing to status lower than the partition requested
-                // increment index so that it's skipped.
-                logDebug(
-                  s"Skipping ${statusPath} as it's lexicographically lesser than" +
-                    s" ${dirPath}")
-                BatchManager.incrementIndex()
-                collect(collected)
-              } else {
-                // We have hit a index which is higher then the one requested for
-                // and also not a child of it.
-                logDebug(
-                  s"Stopped collecting for ${dirPath}." +
-                    s"Encountered non-child ${statusPath}.")
-                collected
-              }
-            }
-
-          case None if !BatchManager.isResourceExhausted =>
-            // Current page is empty, but more pages might be available. Fetch and retry.
-            logInfo(s"Current batch exhausted for ${dirPath}. Fetching next page.")
-            BatchManager.fetchNextPage()
-            collect(collected)
-
-          case _ =>
-            // No current item and the resource is exhausted. We are done.
-            collected
-        }
-      }
-
-      // Initial call to the recursive function. The result is reversed because
-      // prepending (::) is more efficient for lists.
-      // we can reverse it again but that will require space.
-      // We can avoid reversing as listed leaf files can be in any order.
-      val collectedFiles: Seq[FileStatus] = collect(Nil)
-
-      // if there are no leaf files, it might be possible that it's a file
-      val listedFiles = if (collectedFiles.nonEmpty) {
-        collectedFiles
-      } else {
-        val fs = fileSystemProvider(path, hadoopConf)
-        try {
-          logDebug(s"Got nothing from directory listing of ${path}. Checking if it's a file.")
-          Seq(fs.getFileStatus(path))
-        } catch {
-          case e: FileNotFoundException =>
-            logDebug(s"Requested path ${path} is not even a file.")
-            Seq.empty[FileStatus]
-        }
-      }
-
-      logInfo(s"Finished listing for ${path}. Collected ${listedFiles.size} leaf files.")
-      listedFiles.toArray
-    } finally {
-      updateServedPath(path)
-    }
-  }
-
-  // made public only for unit testing
-
-  def canServeFromExistingBatch(path: Path): Boolean = {
-    !lastRequested.isDefined || BatchManager.canServeFromCurrentResponse(path)
-  }
-
-  /**
-   * Heuristic function to decide if using consolidated listing will actually be performant. We
-   * have use lis call base heuristic here i.e. number of calls required to list all partitions.
-   *
-   * Here we predict the number of calls required based on the current batch's score.
-   */
-  def shouldUseConsolidatedListing(
-      currentPath: Path,
-      remainingPath: Int,
-      parallelismThreshold: Int): Boolean = {
-    try {
-      canServeFromExistingBatch(currentPath) || {
-        val score = currentScore()
-        // Safety check to prevent division by zero.
-        // Ideally score should never be zero and always >=1
-        if (score <= 0) {
-          false
-        } else {
-          val estimatedListCalls = remainingPath / score
-          estimatedListCalls <= parallelismThreshold
-        }
-      }
-    } catch {
-      case e: Exception =>
-        logWarning(
-          s"An error occurred while checking for consolidated listing on path " +
-            s"'$currentPath'. Defaulting to false.",
-          e)
-        false
-    }
-  }
-
-  // made public only for unit testing purpose
-  def currentScore(): Int = {
-    BatchManager.fetchBatchScore()
-  }
-
-}
diff --git a/core/src/main/scala/org/apache/spark/util/HadoopFSUtils.scala b/core/src/main/scala/org/apache/spark/util/HadoopFSUtils.scala
index ee254a867b9..bb03cac8f1f 100644
--- a/core/src/main/scala/org/apache/spark/util/HadoopFSUtils.scala
+++ b/core/src/main/scala/org/apache/spark/util/HadoopFSUtils.scala
@@ -20,8 +20,6 @@ package org.apache.spark.util
 import java.io.FileNotFoundException

 import scala.collection.mutable
-import scala.collection.mutable.ListBuffer
-import scala.util.{Failure, Success, Try}

 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs._
@@ -36,10 +34,6 @@ import org.apache.spark.metrics.source.HiveCatalogMetrics
  * Utility functions to simplify and speed-up file listing.
  */
 private[spark] object HadoopFSUtils extends Logging {
-
-  private val partitionListing = (fs: FileSystem, path: Path) => {
-    fs.listStatus(path)
-  }
   /**
    * Lists a collection of paths recursively. Picks the listing strategy adaptively depending
    * on the number of paths to list.
@@ -50,8 +44,6 @@ private[spark] object HadoopFSUtils extends Logging {
    * @param paths Input paths to list
    * @param hadoopConf Hadoop configuration
    * @param filter Path filter used to exclude leaf files from result
-   * @param fileSystemProvider lambda function to get the fileSystem of path.
-   *                           It will help in unit testing as we can mock the filesystem provider.
    * @param ignoreMissingFiles Ignore missing files that happen during recursive listing
    *                           (e.g., due to race conditions)
    * @param ignoreLocality Whether to fetch data locality info when listing leaf files. If false,
@@ -62,7 +54,6 @@ private[spark] object HadoopFSUtils extends Logging {
    * @param parallelismMax The maximum parallelism for listing. If the number of input paths is
    *                       larger than this value, parallelism will be throttled to this value
    *                       to avoid generating too many tasks.
-   * @param consolidatedListing Whether to use consolidated Listing for listing lead files.
    * @return for each input path, the set of discovered files for the path
    */
   def parallelListLeafFiles(
@@ -70,118 +61,12 @@ private[spark] object HadoopFSUtils extends Logging {
     paths: Seq[Path],
     hadoopConf: Configuration,
     filter: PathFilter,
-    fileSystemProvider: (Path, Configuration) => FileSystem,
     ignoreMissingFiles: Boolean,
     ignoreLocality: Boolean,
     parallelismThreshold: Int,
-    parallelismMax: Int,
-    consolidatedListing: Boolean
-    ): Seq[(Path, Seq[FileStatus])] = {
-
-    if (consolidatedListing) {
-      val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-      getConsolidatedListingWithFallback(sc, paths, hadoopConf, filter, isRootLevel = true,
-        ignoreMissingFiles, ignoreLocality, parallelismThreshold, parallelismMax,
-        listingManager, fileSystemProvider)
-    } else {
-      parallelListLeafFilesInternal(sc, paths, hadoopConf, filter,
-        isRootLevel = true, ignoreMissingFiles, ignoreLocality, parallelismThreshold,
-        parallelismMax, fileSystemProvider)
-    }
-  }
-
-  // scalastyle:off argcount
-  private def getConsolidatedListingWithFallback(
-            sc: SparkContext,
-            paths: Seq[Path],
-            hadoopConf: Configuration,
-            filter: PathFilter,
-            isRootLevel: Boolean,
-            ignoreMissingFiles: Boolean,
-            ignoreLocality: Boolean,
-            parallelismThreshold: Int,
-            parallelismMax: Int,
-            listingManager: ConsolidatedListingManager,
-            fileSystemProvider: (Path, Configuration) => FileSystem
-            ): Seq[(Path, Seq[FileStatus])] = {
-
-    val sortedPaths = paths.sorted
-    val pathIterator = sortedPaths.iterator
-    val consolidatedListingFunc = (_: FileSystem, p: Path) => listingManager.listLeafFilesV2(p)
-
-    val consolidatedResults = new ListBuffer[(Path, Seq[FileStatus])]()
-    // Use a var to hold the paths that need the fallback.
-    var remainingPathsForFallback: Seq[Path] = Seq.empty
-    var continueConsolidated = true
-    // keeps track of paths which aren't listed yet
-    var remainingPathsCount: Int = sortedPaths.size
-
-    while (continueConsolidated && pathIterator.hasNext) {
-      val currentPath = pathIterator.next()
-      remainingPathsCount -= 1;
-
-      if (listingManager.shouldUseConsolidatedListing(
-        currentPath,
-        remainingPathsCount,
-        parallelismThreshold)) {
-        Try {
-          listLeafFiles(
-            currentPath,
-            hadoopConf,
-            filter,
-            Some(sc),
-            ignoreMissingFiles,
-            ignoreLocality,
-            isRootLevel,
-            parallelismThreshold,
-            parallelismMax,
-            consolidatedListingFunc,
-            fileSystemProvider)
-        } match {
-          case Success(files) =>
-            consolidatedResults.append((currentPath, files))
-
-          case Failure(e) =>
-            // Using consolidated listing failed, prepare for fallback.
-            logWarning(
-              s"Consolidated listing for path $currentPath failed. " +
-                s"Falling back to parallel listing for remaining paths.",
-              e)
-            continueConsolidated = false
-            // Capture the failed path plus the rest of the iterator.
-            remainingPathsForFallback = currentPath +: pathIterator.toSeq
-        }
-      } else {
-        // can't use consolidated listing anymore, prepare for fallback.
-        continueConsolidated = false
-        // Capture the current path plus the rest of the iterator.
-        remainingPathsForFallback = currentPath +: pathIterator.toSeq
-      }
-    }
-
-    val parallelListingResults = if (remainingPathsForFallback.nonEmpty) {
-      logWarning(
-        s"Listed ${consolidatedResults.size} via consolidated listing." +
-          s" Falling back to parallel listing for ${remainingPathsForFallback.size} paths" +
-          s" starting from path: ${remainingPathsForFallback.head.toString} ")
-
-      parallelListLeafFilesInternal(
-        sc,
-        remainingPathsForFallback,
-        hadoopConf,
-        filter,
-        isRootLevel = true,
-        ignoreMissingFiles,
-        ignoreLocality,
-        parallelismThreshold,
-        parallelismMax,
-        fileSystemProvider)
-    } else {
-      Seq.empty
-    }
-
-    // Combine the results from both stages.
-    consolidatedResults.toSeq ++ parallelListingResults
+    parallelismMax: Int): Seq[(Path, Seq[FileStatus])] = {
+    parallelListLeafFilesInternal(sc, paths, hadoopConf, filter, isRootLevel = true,
+      ignoreMissingFiles, ignoreLocality, parallelismThreshold, parallelismMax)
   }

   private def parallelListLeafFilesInternal(
@@ -193,8 +78,7 @@ private[spark] object HadoopFSUtils extends Logging {
       ignoreMissingFiles: Boolean,
       ignoreLocality: Boolean,
       parallelismThreshold: Int,
-      parallelismMax: Int,
-      fileSystemProvider: (Path, Configuration) => FileSystem): Seq[(Path, Seq[FileStatus])] = {
+      parallelismMax: Int): Seq[(Path, Seq[FileStatus])] = {

     // Short-circuits parallel listing when serial listing is likely to be faster.
     if (paths.size <= parallelismThreshold) {
@@ -208,12 +92,10 @@ private[spark] object HadoopFSUtils extends Logging {
           ignoreLocality = ignoreLocality,
           isRootPath = isRootLevel,
           parallelismThreshold = parallelismThreshold,
-          parallelismMax = parallelismMax,
-          partitionListing,
-          fileSystemProvider)
+          parallelismMax = parallelismMax)
         (path, leafFiles)
       }
-     }
+    }

     logInfo(s"Listing leaf files and directories in parallel under ${paths.length} paths." +
       s" The first several paths are: ${paths.take(10).mkString(", ")}.")
@@ -249,9 +131,7 @@ private[spark] object HadoopFSUtils extends Logging {
               ignoreLocality = ignoreLocality,
               isRootPath = isRootLevel,
               parallelismThreshold = Int.MaxValue,
-              parallelismMax = 0,
-              partitionListing,
-              fileSystemProvider)
+              parallelismMax = 0)
             (path, leafFiles)
           }
         }.collect()
@@ -260,7 +140,7 @@ private[spark] object HadoopFSUtils extends Logging {
     }
   }

-
+  // scalastyle:off argcount
   /**
    * Lists a single filesystem path recursively. If a `SparkContext` object is specified, this
    * function may launch Spark jobs to parallelize listing based on `parallelismThreshold`.
@@ -278,12 +158,10 @@ private[spark] object HadoopFSUtils extends Logging {
       ignoreLocality: Boolean,
       isRootPath: Boolean,
       parallelismThreshold: Int,
-      parallelismMax: Int,
-      listingFunction: (FileSystem, Path) => Array[FileStatus],
-      fileSystemProvider: (Path, Configuration) => FileSystem): Seq[FileStatus] = {
+      parallelismMax: Int): Seq[FileStatus] = {

     logTrace(s"Listing $path")
-    val fs = fileSystemProvider(path, hadoopConf)
+    val fs = path.getFileSystem(hadoopConf)

     // Note that statuses only include FileStatus for the files and dirs directly under path,
     // and does not include anything else recursively.
@@ -299,7 +177,7 @@ private[spark] object HadoopFSUtils extends Logging {
             def next(): LocatedFileStatus = remoteIter.next
             def hasNext(): Boolean = remoteIter.hasNext
           }.toArray
-        case _ => listingFunction(fs, path)
+        case _ => fs.listStatus(path)
       }
     } catch {
       // If we are listing a root path for SQL (e.g. a top level directory of a table), we need to
@@ -342,8 +220,7 @@ private[spark] object HadoopFSUtils extends Logging {
             ignoreMissingFiles = ignoreMissingFiles,
             ignoreLocality = ignoreLocality,
             parallelismThreshold = parallelismThreshold,
-            parallelismMax = parallelismMax,
-            fileSystemProvider = fileSystemProvider
+            parallelismMax = parallelismMax
           ).flatMap(_._2)
         case _ =>
           dirs.flatMap { dir =>
@@ -356,9 +233,7 @@ private[spark] object HadoopFSUtils extends Logging {
               ignoreLocality = ignoreLocality,
               isRootPath = false,
               parallelismThreshold = parallelismThreshold,
-              parallelismMax = parallelismMax,
-              listingFunction = partitionListing,
-              fileSystemProvider = fileSystemProvider)
+              parallelismMax = parallelismMax)
           }
       }
       val filteredTopLevelFiles = if (filter != null) {
diff --git a/core/src/main/scala/org/apache/spark/util/ListingUtil.scala b/core/src/main/scala/org/apache/spark/util/ListingUtil.scala
deleted file mode 100644
index fd7c4eb5c02..00000000000
--- a/core/src/main/scala/org/apache/spark/util/ListingUtil.scala
+++ /dev/null
@@ -1,86 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.spark.util
-
-import java.net.URI
-
-import scala.util.Try
-
-import org.apache.hadoop.fs.{FileSystem, Path}
-
-private[spark] case object ListingUtil {
-
-  def listMethodOptimized(fs: FileSystem): Try[java.lang.reflect.Method] = {
-    Try {
-      val clazz = fs.getClass
-      clazz.getDeclaredMethod("listStatusStartingFrom", classOf[Path])
-    }
-  }
-
-  def invokeMethodJavaReflection[T](
-      method: java.lang.reflect.Method,
-      instance: Any,
-      args: AnyRef*): Try[T] = {
-    Try {
-      method.invoke(instance, args: _*).asInstanceOf[T]
-    }
-  }
-
-  /**
-   * Checks if a child Path is a direct or indirect descendant of a parent Path. This version is
-   * optimized to avoid object creation and correctly compares the full URI, including the scheme
-   * and authority.
-   *
-   * @param parent
-   *   The parent Path.
-   * @param child
-   *   The potential child Path.
-   * @return
-   *   True if the child is a descendant of the parent, false otherwise.
-   */
-  def isChildOf(parent: Path, child: Path): Boolean = {
-    // Get the canonical string representations of the full URIs
-    val parentStr = parent.toUri.toString
-    val childStr = child.toUri.toString
-
-    // A path cannot be a child of itself.
-    if (parentStr == childStr) {
-      return false
-    }
-
-    // Normalize the parent path string to ensure it ends with a slash
-    val normalizedParentStr = if (parentStr.endsWith("/")) {
-      parentStr
-    } else {
-      parentStr + "/"
-    }
-
-    // Perform a direct string comparison
-    childStr.startsWith(normalizedParentStr)
-  }
-
-  def toDir(path: Path): Path = {
-    val uriString = path.toUri.toString
-    if (uriString.endsWith("/")) {
-      path
-    } else {
-      new Path(URI.create(uriString + "/"))
-    }
-  }
-
-}
diff --git a/core/src/test/scala/org/apache/spark/util/ConsolidatedListingManagerSuite.scala b/core/src/test/scala/org/apache/spark/util/ConsolidatedListingManagerSuite.scala
deleted file mode 100644
index c31adb4ea59..00000000000
--- a/core/src/test/scala/org/apache/spark/util/ConsolidatedListingManagerSuite.scala
+++ /dev/null
@@ -1,712 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.spark.util
-
-import java.io.IOException
-
-import org.apache.hadoop.conf.Configuration
-import org.apache.hadoop.fs.{FileStatus, Path}
-import org.mockito.{ArgumentCaptor, ArgumentMatchers, Mockito}
-import org.mockito.Mockito._
-import org.scalatest.matchers.should.Matchers.{a, convertToAnyShouldWrapper}
-
-import org.apache.spark.{SparkConf, SparkFunSuite}
-import org.apache.spark.deploy.SparkHadoopUtil
-import org.apache.spark.util.OverrideListStatusStartingFrom.{convertToFileStatus, convertToFileStatuses}
-
-class ConsolidatedListingManagerSuite extends SparkFunSuite {
-
-  private val sparkConf = new SparkConf()
-  private val hadoopConf = SparkHadoopUtil.newConfiguration(sparkConf)
-
-  test("listing all files in bucket") {
-    val files = Seq(
-      "hdfs://xyz/o1",
-      "hdfs://xyz/o2",
-      "hdfs://xyz/d1/",
-      "hdfs://xyz/d1/o1",
-      "hdfs://xyz/d2/",
-      "hdfs://xyz/d2/o2")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(files)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val paths: Seq[Path] = Seq(new Path("hdfs://xyz"))
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val result = paths.map { path =>
-      val leafFiles = listingManager.listLeafFilesV2(path)
-      (path, leafFiles.toSeq)
-    }
-    verifyListedPaths(result, paths, files)
-    // 1. call to fetch the all the lists
-    // 2. another call to make sure that all resource are actually exhausted
-    verifyListStatusStartingFromCalls(
-      spiedFS,
-      2,
-      Seq(ListingUtil.toDir(new Path("hdfs://xyz")), new Path("hdfs://xyz/o2")))
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-
-  }
-
-  test("listing multiple partitions-LexicographicalOrder A/, B/ ") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file2",
-      "hdfs://xyz/B/file2",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file3")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val paths: Seq[Path] = Seq(new Path("hdfs://xyz/A/"), new Path("hdfs://xyz/B/"))
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val result = paths.map { path =>
-      val leafFiles = listingManager.listLeafFilesV2(path)
-      (path, leafFiles.toSeq)
-    }
-    verifyListedPaths(result, paths, allFiles)
-    // 1. first call will fetch everything > A/
-    // B/ can be served from the same request as B/ > A/
-    verifyListStatusStartingFromCalls(
-      spiedFS,
-      1,
-      Seq(ListingUtil.toDir(new Path("hdfs://xyz/A/"))))
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-    assert(listingManager.currentScore() == 2)
-  }
-
-  test("listing multiple partitions-LexicographicalOrder- skipping some partition A/, C/ ") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file2",
-      "hdfs://xyz/B/file2",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file3",
-      "hdfs://xyz/D/file4")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val paths: Seq[Path] = Seq(new Path("hdfs://xyz/A/"), new Path("hdfs://xyz/C/"))
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val result = paths.map { path =>
-      val leafFiles = listingManager.listLeafFilesV2(path)
-      (path, leafFiles.toSeq)
-    }
-    verifyListedPaths(result, paths, allFiles)
-    // 1. first call will fetch everything > A/
-    // B/ can be served from the same request as B/ > A/
-    verifyListStatusStartingFromCalls(
-      spiedFS,
-      1,
-      Seq(ListingUtil.toDir(new Path("hdfs://xyz/A/"))))
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-    assert(listingManager.currentScore() == 2)
-  }
-
-  test("listing multiple partitions-Non-LexicographicalOrder B/, A/ ") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file2",
-      "hdfs://xyz/B/file2",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file3")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val paths: Seq[Path] = Seq(new Path("hdfs://xyz/B"), new Path("hdfs://xyz/A/"))
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val result = paths.map { path =>
-      val leafFiles = listingManager.listLeafFilesV2(path)
-      (path, leafFiles.toSeq)
-    }
-    verifyListedPaths(result, paths, allFiles)
-    // 1. call to fetch the everything > B/
-    // 2. another call to fetch everything >A/ as B/ will not be able to satisfy
-    verifyListStatusStartingFromCalls(
-      spiedFS,
-      2,
-      Seq(
-        ListingUtil.toDir(new Path("hdfs://xyz/B")),
-        ListingUtil.toDir(new Path("hdfs://xyz/A/"))))
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-  }
-
-  test("listing multiple partitions-Non-LexicographicalOrder A/, A/B/ ") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file2",
-      "hdfs://xyz/B/file2",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file3")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val paths: Seq[Path] = Seq(new Path("hdfs://xyz/A"), new Path("hdfs://xyz/A/B"))
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val result = paths.map { path =>
-      val leafFiles = listingManager.listLeafFilesV2(path)
-      (path, leafFiles.toSeq)
-    }
-    verifyListedPaths(result, paths, allFiles)
-    // 1. call to fetch the everything > A/
-    // as A/B is a subdirectory of already listed A/, we need to fetch fresh again for A/B
-    verifyListStatusStartingFromCalls(
-      spiedFS,
-      2,
-      Seq(
-        ListingUtil.toDir(new Path("hdfs://xyz/A")),
-        ListingUtil.toDir(new Path("hdfs://xyz/A/B"))))
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-  }
-
-  test("listing partitions and files-LexicographicalOrder A/, A/B/file") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file1",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file3")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val dirPath: Path = new Path("hdfs://xyz/A")
-
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val leafFiles = listingManager.listLeafFilesV2(dirPath)
-
-    verifyListedPath(dirPath, leafFiles, allFiles)
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(dirPath)))
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-
-    Mockito.reset(spiedFS)
-    val filePath: Path = new Path("hdfs://xyz/A/B/file1")
-    val fileStatus = listingManager.listLeafFilesV2(filePath)
-    // current index will be pointing at C/, can't be served from existing batch
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(filePath)))
-    // well it's not a directory, hence fileStatus call
-    verifyGetFileStatusCalls(spiedFS, 1, Seq(filePath))
-    verifyListedPath(filePath, fileStatus, allFiles)
-    assert(listingManager.currentScore() == 1)
-  }
-
-  test("listing partitions and files-Non-LexicographicalOrder-A/B/file, A/") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file1",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file3")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-
-    val filePath: Path = new Path("hdfs://xyz/A/B/file1")
-    val fileStatus = listingManager.listLeafFilesV2(filePath)
-
-    verifyListedPath(filePath, fileStatus, allFiles)
-    // list call assuming it's a directory, and it is a directory
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(filePath)))
-    // well it's not a directory, hence fileStatus call
-    verifyGetFileStatusCalls(spiedFS, 1, Seq(filePath))
-
-    Mockito.reset(spiedFS)
-
-    val dirPath: Path = new Path("hdfs://xyz/A")
-    val leafFiles = listingManager.listLeafFilesV2(dirPath)
-    // existing batch although contains data but as the order of path request is wrong
-    // it will make a fresh batch request
-    // existing batch contains everything >= A/B/file1/
-    // for listing everything > A/ it needs to make a fresh batch request
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(dirPath)))
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-
-    verifyListedPath(dirPath, leafFiles, allFiles)
-  }
-
-  test("listing partitions and files-LexicographicalOrder A/, B/file") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file1",
-      "hdfs://xyz/B/file2",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file3")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val paths: Seq[Path] = Seq(new Path("hdfs://xyz/A"), new Path("hdfs://xyz/B/file2"))
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val result = paths.map { path =>
-      val leafFiles = listingManager.listLeafFilesV2(path)
-      (path, leafFiles.toSeq)
-    }
-    verifyListedPaths(result, paths, allFiles)
-    // Only one call as another file will be listed due to consolidated listing
-    verifyListStatusStartingFromCalls(
-      spiedFS,
-      1,
-      Seq(ListingUtil.toDir(new Path("hdfs://xyz/A"))))
-    assert(listingManager.currentScore() == 2)
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-  }
-
-  test("listing partitions and files-Non-LexicographicalOrder B/file, A/") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file1",
-      "hdfs://xyz/B/file2",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file3")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val filePath: Path = new Path("hdfs://xyz/B/file2")
-    val dirPath: Path = new Path("hdfs://xyz/A")
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val fileStatus = listingManager.listLeafFilesV2(filePath)
-    verifyListedPath(filePath, fileStatus, allFiles)
-    // list call assuming it's a directory
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(filePath)))
-    //  FileStatus call as it's a file
-    verifyGetFileStatusCalls(spiedFS, 1, Seq(filePath))
-
-    Mockito.reset(spiedFS)
-
-    val leafFiles = listingManager.listLeafFilesV2(dirPath)
-    // list call assuming it's a directory
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(dirPath)))
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-    verifyListedPath(dirPath, leafFiles, allFiles)
-  }
-
-  test("listing partitions and files- file is part of partition listed") {
-    val files = Seq(
-      "hdfs://xyz/d1/",
-      "hdfs://xyz/d1/o1",
-      "hdfs://xyz/d1/sub-d1/o1",
-      "hdfs://xyz/d2/",
-      "hdfs://xyz/d2/sub-d2/o2")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(files)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    var paths: Seq[Path] = Seq(new Path("hdfs://xyz/d1"), new Path("hdfs://xyz/d1/sub-d1/o1"))
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    var result = paths.map { path =>
-      val leafFiles = listingManager.listLeafFilesV2(path)
-      (path, leafFiles.toSeq)
-    }
-    verifyListedPaths(result, paths, files)
-    // Two list calls
-    // One where all partitions after d1/ were listed
-    // another sub-d1/ is child of previous one so another list call
-    // to start listing everything after this sub partition
-    verifyListStatusStartingFromCalls(
-      spiedFS,
-      2,
-      Seq(
-        ListingUtil.toDir(new Path("hdfs://xyz/d1")),
-        ListingUtil.toDir(new Path("hdfs://xyz/d1/sub-d1/o1"))))
-    // given it's a file and we as always assume it to be a directory during listing,
-    // will not fetch the status by listing
-    // we then fallback to assuming it to be a file and call getFileStatus
-    verifyGetFileStatusCalls(spiedFS, 1, Seq(new Path("hdfs://xyz/d1/sub-d1/o1")))
-
-    Mockito.reset(spiedFS)
-
-    paths = Seq(
-      new Path("hdfs://xyz/d1"),
-      new Path("hdfs://xyz/d1/sub-d1/I-dont-exist") // <-- non existent file
-    )
-
-    result = paths.map { path =>
-      val leafFiles = listingManager.listLeafFilesV2(path)
-      (path, leafFiles.toSeq)
-    }
-
-    verifyListedPaths(result, paths, files)
-    // Two list calls
-    // One where all partitions after d1/ were listed
-    // another sub-d1/ is child of previous one so another list call
-    // to start listing everything after this sub partition
-    verifyListStatusStartingFromCalls(
-      spiedFS,
-      2,
-      Seq(
-        ListingUtil.toDir(new Path("hdfs://xyz/d1")),
-        ListingUtil.toDir(new Path("hdfs://xyz/d1/sub-d1/I-dont-exist"))))
-    // given it's a file and we as always assume it to be a directory during listing,
-    // will not fetch the status by listing
-    // we then fallback to assuming it to be a file and call getFileStatus
-    verifyGetFileStatusCalls(spiedFS, 1, Seq(new Path("hdfs://xyz/d1/sub-d1/I-dont-exist")))
-
-  }
-
-  test("listing file- file don't exists") {
-    val files = Seq(
-      "hdfs://xyz/d1/",
-      "hdfs://xyz/d1/o1",
-      "hdfs://xyz/d1/sub-d1/o1",
-      "hdfs://xyz/d2/",
-      "hdfs://xyz/d2/sub-d2/o2")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(files)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val path: Path = new Path("hdfs://xyz/d1/sub-d1/I-dont-exist")
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val result: Seq[FileStatus] = listingManager.listLeafFilesV2(path)
-    // One list call, as we assume it to be a directory,
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(path)))
-    // Given it's file listing directory will not fetch status
-    // So, we make getFileStatus call
-    verifyGetFileStatusCalls(spiedFS, 1, Seq(path))
-
-    assert(result.size == 0)
-  }
-
-  test("listing partition - partition don't exist") {
-    val files = Seq(
-      "hdfs://xyz/d1/",
-      "hdfs://xyz/d1/o1",
-      "hdfs://xyz/d1/sub-d1/o1",
-      "hdfs://xyz/d2/",
-      "hdfs://xyz/d2/sub-d2/o2")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(files)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val path: Path = new Path("hdfs://xyz/d1/I-dont-exist/")
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val result: Seq[FileStatus] = listingManager.listLeafFilesV2(path)
-    // One list call, as we assume it to be a directory,
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(path)))
-    // Given it is a file, listing directory will not fetch status
-    // So, we make getFileStatus call
-    verifyGetFileStatusCalls(spiedFS, 1, Seq(path))
-    assert(result.size == 0)
-  }
-
-  test("listing partition - multiple pages") {
-    val allFiles = Seq(
-      "hdfs://xyz/d1/",
-      "hdfs://xyz/d1/o1",
-      "hdfs://xyz/d1/sub-d1/sdo1",
-      "hdfs://xyz/d1/sub-d1/sdo2",
-      "hdfs://xyz/d1/sub-d1/sdo3",
-      "hdfs://xyz/d1/sub-d1/sdo4",
-      "hdfs://xyz/d1/sub-d1/sdo5",
-      "hdfs://xyz/d1/sub-d1/sdo6",
-      "hdfs://xyz/d1/sub-d1/sdo7",
-      "hdfs://xyz/d1/sub-d1/sdo8",
-      "hdfs://xyz/d1/sub-d1/sdo9",
-      "hdfs://xyz/d1/sub-d1/sdo10",
-      "hdfs://xyz/d2/",
-      "hdfs://xyz/d2/sub-d2/o2")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val path: Path = new Path("hdfs://xyz/d1/")
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val leafFiles: Seq[FileStatus] = listingManager.listLeafFilesV2(path)
-    // One list call, as we assume it to be a directory,
-    verifyListStatusStartingFromCalls(
-      spiedFS,
-      2,
-      Seq(ListingUtil.toDir(path), new Path("hdfs://xyz/d1/sub-d1/sdo8")))
-    verifyGetFileStatusCalls(spiedFS, 0, Seq())
-    verifyListedPath(path, leafFiles, allFiles)
-    assert(leafFiles.size == 11)
-  }
-
-  // ======== ListingManager canServeFromExistingBatch ======
-
-  test("ListingManager canServeFromExistingBatch-List A/, check for A/B, /B") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file1",
-      "hdfs://xyz/B-file0",
-      "hdfs://xyz/B/",
-      "hdfs://xyz/B/file2",
-      "hdfs://xyz/B/C/file3",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file4")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val filePath: Path = new Path("hdfs://xyz/B/file2")
-    // val dirPath: Path = new Path("hdfs://xyz/A/B")
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val leafFiles = listingManager.listLeafFilesV2(filePath)
-    verifyListedPath(filePath, leafFiles, allFiles)
-
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(filePath)))
-    verifyGetFileStatusCalls(spiedFS, 1, Seq(filePath))
-
-    val subDir: Path = new Path("hdfs://xyz/B/C")
-    assert(listingManager.canServeFromExistingBatch(subDir) == false)
-
-    val lexicographicallyLowerPath = new Path("hdfs://xyz/B-file0")
-    assert(listingManager.canServeFromExistingBatch(lexicographicallyLowerPath) == false)
-
-    val backwardSeekPath = new Path("hdfs://xyz/A")
-    assert(listingManager.canServeFromExistingBatch(backwardSeekPath) == false)
-
-    val forwardSeekPath = new Path("hdfs://xyz/C")
-    assert(listingManager.canServeFromExistingBatch(forwardSeekPath) == true)
-  }
-
-  test("ListingManager page score test") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file1",
-      "hdfs://xyz/B-file0",
-      "hdfs://xyz/B/",
-      "hdfs://xyz/B/file2",
-      "hdfs://xyz/B/C/file3",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file4")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val dirPath: Path = new Path("hdfs://xyz/A")
-    // val dirPath: Path = new Path("hdfs://xyz/A/B")
-    val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-    val leafFiles = listingManager.listLeafFilesV2(dirPath)
-    verifyListedPath(dirPath, leafFiles, allFiles)
-
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(ListingUtil.toDir(dirPath)))
-
-    var testPath: Path = new Path("hdfs://xyz/B-file0")
-    var testLeafFiles = listingManager.listLeafFilesV2(testPath)
-    verifyListedPath(testPath, testLeafFiles, allFiles)
-    assert(listingManager.currentScore() == 2)
-
-    testPath = new Path("hdfs://xyz/B/")
-    testLeafFiles = listingManager.listLeafFilesV2(testPath)
-    verifyListedPath(testPath, testLeafFiles, allFiles)
-    assert(listingManager.currentScore() == 3)
-
-    Mockito.reset(spiedFS)
-    testPath = new Path("hdfs://xyz/C")
-    testLeafFiles = listingManager.listLeafFilesV2(testPath)
-    verifyListedPath(testPath, testLeafFiles, allFiles)
-    // although it's served from the current batch
-    // another request is made to make sure no other files are there as it reached end of batch
-    // latest batch will point towards exhausted batch.
-    verifyListStatusStartingFromCalls(spiedFS, 1, Seq(new Path("hdfs://xyz/C/D/file4")))
-    assert(listingManager.currentScore() == 1)
-  }
-
-  test("ListingManager- ListStatusStartingFrom doesn't exist") {
-    withTempDir { dir =>
-      val dirPath = new Path(dir.getAbsolutePath)
-      val fileSystemProvider = (path: Path, config: Configuration) => {
-        path.getFileSystem(config)
-      }
-
-      val listingManager = new ConsolidatedListingManager(hadoopConf, fileSystemProvider)
-      val exception = intercept[IOException] {
-        listingManager.listLeafFilesV2(dirPath)
-      }
-
-      exception shouldBe a[IOException]
-
-      val cause = exception.getCause
-      cause shouldBe a[NoSuchMethodException]
-    }
-
-  }
-
-  private def verifyListStatusStartingFromCalls(
-      mockedFS: OverrideListStatusStartingFrom,
-      expectedInvocationCount: Int,
-      expectedPath: Seq[Path]): Unit = {
-    assert(expectedInvocationCount == expectedPath.size)
-    if (expectedInvocationCount == 0) {
-      verify(mockedFS, times(expectedInvocationCount))
-        .listStatusStartingFrom(ArgumentMatchers.any[Path])
-    } else {
-      val pathCaptor = ArgumentCaptor.forClass(classOf[Path])
-      verify(mockedFS, times(expectedInvocationCount)).listStatusStartingFrom(
-        pathCaptor.capture())
-      import scala.collection.JavaConverters._
-      val capturedPaths: Seq[Path] = pathCaptor.getAllValues.asScala
-      assert(capturedPaths.length == expectedPath.size)
-      for (i <- expectedPath.indices) {
-        assert(expectedPath(i).toUri.getPath.equals(capturedPaths(i).toUri.getPath))
-
-      }
-    }
-  }
-
-  private def verifyGetFileStatusCalls(
-      mockedFS: OverrideListStatusStartingFrom,
-      expectedInvocationCount: Int,
-      expectedPath: Seq[Path]): Unit = {
-    assert(expectedInvocationCount == expectedPath.size)
-    if (expectedInvocationCount == 0) {
-      verify(mockedFS, times(expectedInvocationCount))
-        .getFileStatus(ArgumentMatchers.any[Path])
-    } else {
-      val pathCaptor = ArgumentCaptor.forClass(classOf[Path])
-      verify(mockedFS, times(expectedInvocationCount)).getFileStatus(pathCaptor.capture())
-      import scala.collection.JavaConverters._
-      val capturedPaths: Seq[Path] = pathCaptor.getAllValues.asScala
-      assert(capturedPaths.length == expectedPath.size)
-      for (i <- expectedPath.indices) {
-        assert(expectedPath(i).toString.equals(capturedPaths(i).toString))
-
-      }
-    }
-  }
-
-  private def expectedLeafFiles(path: Path, allFiles: Seq[String]): Seq[FileStatus] = {
-    val file = allFiles.find(file => file.compareTo(path.toString) == 0)
-    file match {
-      case Some(file) =>
-        Seq(convertToFileStatus(file))
-      case None =>
-        val dirPath = ListingUtil.toDir(path)
-        val leafFiles = allFiles.filter(f => !f.endsWith("/") && f.startsWith(dirPath.toString))
-        convertToFileStatuses(leafFiles.sorted)
-    }
-  }
-
-  private def verifyListedPaths(
-      result: Seq[(Path, Seq[FileStatus])],
-      paths: Seq[Path],
-      allFiles: Seq[String]): Unit = {
-    assert(result.size == paths.size)
-    paths.foreach(path => {
-      val resultMap = result.toMap
-      assert(resultMap.contains(path))
-      assert(resultMap.get(path).isDefined)
-      verifyListedPath(path, resultMap.get(path).get, allFiles)
-    })
-  }
-
-  private def verifyListedPath(
-      path: Path,
-      result: Seq[FileStatus],
-      allFiles: Seq[String]): Unit = {
-    val leafFiles = expectedLeafFiles(path, allFiles)
-    assert(result.size == leafFiles.length)
-    verifyStatusOrder(result, leafFiles)
-  }
-
-  private def verifyStatusOrder(actual: Seq[FileStatus], expected: Seq[FileStatus]): Unit = {
-    assert(actual.size == expected.size)
-    val expectedPath: Seq[String] = expected.map(s => s.getPath.toString)
-    actual.foreach(status => assert(expectedPath.contains(status.getPath.toString)))
-  }
-
-}
diff --git a/core/src/test/scala/org/apache/spark/util/CustomFileSystem.scala b/core/src/test/scala/org/apache/spark/util/CustomFileSystem.scala
deleted file mode 100644
index 894536c3707..00000000000
--- a/core/src/test/scala/org/apache/spark/util/CustomFileSystem.scala
+++ /dev/null
@@ -1,140 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.spark.util
-
-import java.io.{FileNotFoundException, IOException}
-import java.net.URI
-
-import org.apache.hadoop.fs.{FileStatus, FileSystem, FSDataInputStream, FSDataOutputStream, Path}
-import org.apache.hadoop.fs.permission.FsPermission
-import org.apache.hadoop.util.Progressable
-
-import org.apache.spark.util.OverrideListStatusStartingFrom.{convertToFileStatus, expectedResponseFromListStatus, expectedResponseWithStartFrom}
-
-/**
- * A stub FileSystem for testing purposes. Most methods are unsupported. As this was used
- * specifically for testing listing.
- */
-class CustomFileSystem extends FileSystem {
-  @throws(classOf[IOException])
-  def listStatusStartingFrom(path: Path): Array[FileStatus] = {
-    return Array.empty
-  }
-
-  override def getUri: URI = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def open(path: Path, i: Int): FSDataInputStream = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def create(
-      path: Path,
-      fsPermission: FsPermission,
-      b: Boolean,
-      i: Int,
-      i1: Short,
-      l: Long,
-      progressable: Progressable): FSDataOutputStream = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def append(path: Path, i: Int, progressable: Progressable): FSDataOutputStream = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def rename(path: Path, path1: Path): Boolean = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def delete(path: Path, b: Boolean): Boolean = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def listStatus(path: Path): Array[FileStatus] = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def setWorkingDirectory(path: Path): Unit = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def getWorkingDirectory: Path = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def mkdirs(path: Path, fsPermission: FsPermission): Boolean = {
-    throw new UnsupportedOperationException()
-  }
-
-  override def getFileStatus(path: Path): FileStatus = {
-    throw new UnsupportedOperationException()
-  }
-}
-class OverrideListStatusStartingFrom(files: Seq[String]) extends CustomFileSystem {
-
-  val maxResultSize: Int = 10;
-
-  override def listStatusStartingFrom(path: Path): Array[FileStatus] = {
-    expectedResponseWithStartFrom(path, files).take(maxResultSize)
-  }
-
-  override def listStatus(path: Path): Array[FileStatus] = {
-    expectedResponseFromListStatus(path, files).take(maxResultSize)
-  }
-
-  override def getFileStatus(path: Path): FileStatus = {
-    val res = files.find(f => f.compareTo(path.toString) == 0)
-    return if (res.isDefined) {
-      convertToFileStatus(res.get)
-    } else {
-      throw new FileNotFoundException(s"Path ${path} don't exists")
-    }
-  }
-}
-
-object OverrideListStatusStartingFrom {
-
-  /** Factory method for creating an instance easily. */
-  def apply(files: Seq[String]): OverrideListStatusStartingFrom = {
-    new OverrideListStatusStartingFrom(files)
-  }
-  def convertToFileStatus(file: String): FileStatus = {
-    val isDir = file.endsWith("/")
-    new FileStatus(10, isDir, 1, 1, 0, new Path(file))
-  }
-
-  def convertToFileStatuses(files: Seq[String]): Array[FileStatus] = {
-    files.map(convertToFileStatus).toArray
-  }
-
-  def expectedResponseWithStartFrom(path: Path, allFiles: Seq[String]): Array[FileStatus] = {
-    val pathString = path.toString
-    val filteredFiles = allFiles.filter(f => f.compareTo(pathString) >= 0 && !f.endsWith("/"))
-    val sortedFiles = filteredFiles.sorted
-    convertToFileStatuses(sortedFiles)
-  }
-
-  def expectedResponseFromListStatus(path: Path, allFiles: Seq[String]): Array[FileStatus] = {
-    val pathString = path.toString
-    val filteredFiles = allFiles.filter(f => f.startsWith(pathString) && !f.endsWith("/"))
-    val sortedFiles = filteredFiles.sorted
-    convertToFileStatuses(sortedFiles)
-  }
-}
diff --git a/core/src/test/scala/org/apache/spark/util/HadoopFSUtilsSuite.scala b/core/src/test/scala/org/apache/spark/util/HadoopFSUtilsSuite.scala
index 5acb7260095..ba91eabc1ca 100644
--- a/core/src/test/scala/org/apache/spark/util/HadoopFSUtilsSuite.scala
+++ b/core/src/test/scala/org/apache/spark/util/HadoopFSUtilsSuite.scala
@@ -17,23 +17,9 @@

 package org.apache.spark.util

-import java.io.IOException
-
-import org.apache.hadoop.conf.Configuration
-import org.apache.hadoop.fs.{FileStatus, FileSystem, Path}
-import org.apache.hadoop.mapred.{FileInputFormat, JobConf}
-import org.mockito.ArgumentMatchers
-import org.mockito.Mockito._
-
-import org.apache.spark.{SparkConf, SparkContext, SparkFunSuite}
-import org.apache.spark.deploy.SparkHadoopUtil
-import org.apache.spark.util.OverrideListStatusStartingFrom.{convertToFileStatus, convertToFileStatuses, expectedResponseFromListStatus, expectedResponseWithStartFrom}
+import org.apache.spark.SparkFunSuite

 class HadoopFSUtilsSuite extends SparkFunSuite {
-
-  private val sparkConf = new SparkConf()
-  private val hadoopConf = SparkHadoopUtil.newConfiguration(sparkConf)
-
   test("HadoopFSUtils - file filtering") {
     assert(!HadoopFSUtils.shouldFilterOutPathName("abcd"))
     assert(HadoopFSUtils.shouldFilterOutPathName(".ab"))
@@ -44,197 +30,4 @@ class HadoopFSUtilsSuite extends SparkFunSuite {
     assert(HadoopFSUtils.shouldFilterOutPathName("_cd_common_metadata"))
     assert(HadoopFSUtils.shouldFilterOutPathName("a._COPYING_"))
   }
-
-  test("Consolidated listing- partitions and files") {
-    val allFiles = Seq(
-      "hdfs://xyz/A/",
-      "hdfs://xyz/A/file1",
-      "hdfs://xyz/A/B/file2",
-      "hdfs://xyz/B/file2",
-      "hdfs://xyz/C/",
-      "hdfs://xyz/C/D/file3")
-
-    val hadoopConf = new Configuration(false)
-    val fileSystem = OverrideListStatusStartingFrom(allFiles)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val paths: Seq[Path] = Seq(new Path("hdfs://xyz/A"), new Path("hdfs://xyz/B"))
-    val result = hadoopFSParallelListing(paths, fileSystemProvider)
-    verifyListedPaths(result, paths, allFiles)
-    // Only one call as another file will be listed due to consolidated listing
-
-    verify(spiedFS, times(1)).listStatusStartingFrom(ArgumentMatchers.any[Path])
-  }
-
-  test("Consolidated listing-partitions and files-file is part of partition listed") {
-    val files = Seq(
-      "hdfs://xyz/d1/",
-      "hdfs://xyz/d1/o1",
-      "hdfs://xyz/d1/sub-d1/o1",
-      "hdfs://xyz/d2/",
-      "hdfs://xyz/d2/sub-d2/o2")
-
-    val fileSystem = OverrideListStatusStartingFrom(files)
-    val spiedFS = spy[OverrideListStatusStartingFrom](fileSystem)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      spiedFS
-    }
-
-    val paths: Seq[Path] = Seq(new Path("hdfs://xyz/d1"), new Path("hdfs://xyz/d1/sub-d1/o1"))
-    val result = hadoopFSParallelListing(paths, fileSystemProvider)
-    verifyListedPaths(result, paths, files)
-
-    verify(spiedFS, times(2)).listStatusStartingFrom(ArgumentMatchers.any[Path])
-    verify(spiedFS, times(1)).getFileStatus(ArgumentMatchers.any[Path])
-  }
-
-  test("Sequential Listing- consolidatedListing not implemented") {
-    val files = Seq(
-      "hdfs://xyz/d1/",
-      "hdfs://xyz/d1/o1",
-      "hdfs://xyz/d1/sub-d1/o1",
-      "hdfs://xyz/d2/",
-      "hdfs://xyz/d2/sub-d2/o2")
-
-    val path = new Path("hdfs://xyz/d1")
-    val listStatusResponse = expectedResponseFromListStatus(path, files)
-    val mockFS: FileSystem = mock(classOf[FileSystem])
-
-    when(mockFS.listStatus(ArgumentMatchers.any[Path])).thenReturn(listStatusResponse)
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      mockFS
-    }
-
-    val paths: Seq[Path] = Seq(path)
-    val result = hadoopFSParallelListing(paths, fileSystemProvider)
-    verifyListedPaths(result, paths, files)
-
-    verify(mockFS, times(1)).listStatus(ArgumentMatchers.any[Path])
-  }
-
-  test("Sequential listing- consolidatedListing failure") {
-    val files = Seq(
-      "hdfs://xyz/d1/",
-      "hdfs://xyz/d1/o1",
-      "hdfs://xyz/d1/sub-d1/o1",
-      "hdfs://xyz/d2/",
-      "hdfs://xyz/d2/sub-d2/o2")
-
-    val path = new Path("hdfs://xyz/d1")
-    val listStatusResponse = expectedResponseFromListStatus(path, files)
-
-    val mockFS: OverrideListStatusStartingFrom = mock(classOf[OverrideListStatusStartingFrom])
-
-    // make sure that it throws
-    when(mockFS.listStatusStartingFrom(ArgumentMatchers.any[Path]))
-      .thenThrow(new IOException())
-
-    when(mockFS.listStatus(ArgumentMatchers.any[Path])).thenReturn(listStatusResponse)
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      mockFS
-    }
-
-    val paths: Seq[Path] = Seq(path)
-    val result = hadoopFSParallelListing(paths, fileSystemProvider)
-    verifyListedPaths(result, paths, files)
-
-    // It did tried for consolidated listing
-    verify(mockFS, times(1)).listStatusStartingFrom(ArgumentMatchers.any[Path])
-    // fall back to default
-    verify(mockFS, times(1)).listStatus(ArgumentMatchers.any[Path])
-  }
-
-  test("listing via both - Consolidated and Sequential") {
-    val files = Seq(
-      "hdfs://xyz/d1/",
-      "hdfs://xyz/d1/o1",
-      "hdfs://xyz/d1/sub-d1/o1",
-      "hdfs://xyz/d2/",
-      "hdfs://xyz/d2/sub-d2/o2")
-
-    val path1 = new Path("hdfs://xyz/d1/")
-    val path2 = new Path("hdfs://xyz/d1/sub-d1/")
-    val startFromResponse = expectedResponseWithStartFrom(path1, files)
-    val listStatusResponse = expectedResponseFromListStatus(path2, files)
-
-    val mockFS: OverrideListStatusStartingFrom = mock(classOf[OverrideListStatusStartingFrom])
-
-    when(mockFS.listStatusStartingFrom(ArgumentMatchers.any[Path]))
-      .thenReturn(startFromResponse)
-      .thenThrow(new IOException())
-    when(mockFS.listStatus(ArgumentMatchers.any[Path])).thenReturn(listStatusResponse)
-
-    val fileSystemProvider = (_: Path, _: Configuration) => {
-      mockFS
-    }
-
-    val paths: Seq[Path] = Seq(path1, path2)
-    val result = hadoopFSParallelListing(paths, fileSystemProvider)
-    verifyListedPaths(result, paths, files)
-
-    // First call succeeded, second failed and fallen back to default listing
-    verify(mockFS, times(2)).listStatusStartingFrom(ArgumentMatchers.any[Path])
-    verify(mockFS, times(1)).listStatus(ArgumentMatchers.any[Path])
-  }
-
-  private def hadoopFSParallelListing(
-      paths: Seq[Path],
-      fileSystemProvider: (Path, Configuration) => FileSystem): Seq[(Path, Seq[FileStatus])] = {
-    HadoopFSUtils.parallelListLeafFiles(
-      sc = mock(classOf[SparkContext]), // Use the shared SparkContext
-      paths = paths,
-      hadoopConf = hadoopConf,
-      filter = FileInputFormat.getInputPathFilter(new JobConf(hadoopConf, this.getClass)),
-      ignoreMissingFiles = true,
-      ignoreLocality = true,
-      parallelismThreshold = 10,
-      parallelismMax = 1,
-      consolidatedListing = true,
-      fileSystemProvider = fileSystemProvider)
-  }
-
-  private def expectedLeafFiles(path: Path, allFiles: Seq[String]): Seq[FileStatus] = {
-    val file = allFiles.find(file => file.compareTo(path.toString) == 0)
-    file match {
-      case Some(file) =>
-        Seq(convertToFileStatus(file))
-      case None =>
-        val dirPath = ListingUtil.toDir(path)
-        val leafFiles = allFiles.filter(f => !f.endsWith("/") && f.startsWith(dirPath.toString))
-        convertToFileStatuses(leafFiles.sorted)
-    }
-  }
-
-  private def verifyListedPaths(
-      result: Seq[(Path, Seq[FileStatus])],
-      paths: Seq[Path],
-      allFiles: Seq[String]): Unit = {
-    assert(result.size == paths.size)
-    paths.foreach(path => {
-      val resultMap = result.toMap
-      assert(resultMap.contains(path))
-      assert(resultMap.get(path).isDefined)
-      verifyListedPath(path, resultMap.get(path).get, allFiles)
-    })
-  }
-
-  private def verifyListedPath(
-      path: Path,
-      result: Seq[FileStatus],
-      allFiles: Seq[String]): Unit = {
-    val leafFiles = expectedLeafFiles(path, allFiles)
-    assert(result.size == leafFiles.length)
-    verifyStatusOrder(result, leafFiles)
-  }
-
-  private def verifyStatusOrder(actual: Seq[FileStatus], expected: Seq[FileStatus]): Unit = {
-    assert(actual.size == expected.size)
-    val expectedPath: Seq[String] = expected.map(s => s.getPath.toString)
-    actual.foreach(status => assert(expectedPath.contains(status.getPath.toString)))
-  }
 }
diff --git a/core/src/test/scala/org/apache/spark/util/ListingUtilSuite.scala b/core/src/test/scala/org/apache/spark/util/ListingUtilSuite.scala
deleted file mode 100644
index f145e03dfed..00000000000
--- a/core/src/test/scala/org/apache/spark/util/ListingUtilSuite.scala
+++ /dev/null
@@ -1,58 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.spark.util
-
-import org.apache.hadoop.fs.Path
-
-import org.apache.spark.SparkFunSuite
-
-class ListingUtilSuite extends SparkFunSuite {
-
-  test("verify is child of") {
-    assert(
-      !ListingUtil.isChildOf(
-        /* parent */ new Path("gs://sf10000_parquet/store"),
-        /* child */ new Path("gs://sf10000_parquet/store_sales")))
-
-    assert(
-      ListingUtil.isChildOf(
-        /* parent */ new Path("gs://sf10000_parquet/store"),
-        /* child */ new Path("gs://sf10000_parquet/store/file1")))
-
-    assert(
-      ListingUtil.isChildOf(
-        /* parent */ new Path("gs://sf10000_parquet/store"),
-        /* child */ new Path("gs://sf10000_parquet/store/sub-dir/file2")))
-  }
-
-  test("convert to Dir") {
-    val path: Path =
-      new Path("gs://l/tpch/sf10000/parquet/partitioned/sf10000_parquet/part/p_brand=Brand%2355")
-
-    assert(ListingUtil.toDir(path).toString.endsWith("/"))
-    assert(ListingUtil.toDir(path).toString.startsWith(path.toString))
-
-    val dirPath: Path =
-      new Path("gs://l/tpch/sf10000/parquet/partitioned/sf10000_parquet/part/p_brand=Brand%2355/")
-
-    assert(ListingUtil.toDir(path).toString.startsWith(dirPath.toString))
-    assert(ListingUtil.toDir(path).toString.endsWith("/"))
-    assert(!ListingUtil.toDir(path).toString.endsWith("//"))
-  }
-
-}
diff --git a/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala b/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala
index cb46575c3d0..7121b5ec9ae 100644
--- a/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala
+++ b/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala
@@ -4824,14 +4824,6 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)

-  val CONSOLIDATED_LISTING =
-    buildConf("spark.dataproc.consolidatedListing")
-      .doc("If true, Will use the consolidated listing to fetch leaf files.")
-      .version("3.5.3")
-      .internal()
-      .booleanConf
-      .createWithDefault(false)
-
   val DATAPROC_REWRITE_EXISTS_AS_SCALAR_PROJECT_ENABLED =
     buildConf("spark.dataproc.sql.rewrite.exists.as.scalar.project.enabled")
       .doc("If enabled, rewrites exists subquery to add a scalar project.")
@@ -5422,8 +5414,6 @@ class SQLConf extends Serializable with Logging with SqlApiConf {
   def parallelPartitionDiscoveryParallelism: Int =
     getConf(SQLConf.PARALLEL_PARTITION_DISCOVERY_PARALLELISM)

-  def consolidatedListingEnabled: Boolean = getConf(SQLConf.CONSOLIDATED_LISTING)
-
   def bucketingEnabled: Boolean = getConf(SQLConf.BUCKETING_ENABLED)

   def bucketingMaxBuckets: Int = getConf(SQLConf.BUCKETING_MAX_BUCKETS)
diff --git a/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala b/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala
index 69ee63965ff..44d31131e9c 100644
--- a/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala
+++ b/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala
@@ -150,22 +150,16 @@ object InMemoryFileIndex extends Logging {
       filter: PathFilter,
       sparkSession: SparkSession,
       parameters: Map[String, String] = Map.empty): Seq[(Path, Seq[FileStatus])] = {
-    val fileSystemProvider = (path: Path, hadoopConf: Configuration) => {
-      path.getFileSystem(hadoopConf)
-    }
     HadoopFSUtils.parallelListLeafFiles(
       sc = sparkSession.sparkContext,
       paths = paths,
       hadoopConf = hadoopConf,
       filter = new PathFilterWrapper(filter),
-      fileSystemProvider = fileSystemProvider,
       ignoreMissingFiles =
         new FileSourceOptions(CaseInsensitiveMap(parameters)).ignoreMissingFiles,
       ignoreLocality = sparkSession.sessionState.conf.ignoreDataLocality,
       parallelismThreshold = sparkSession.sessionState.conf.parallelPartitionDiscoveryThreshold,
-      parallelismMax = sparkSession.sessionState.conf.parallelPartitionDiscoveryParallelism,
-      consolidatedListing = sparkSession.sessionState.conf.consolidatedListingEnabled
-    )
+      parallelismMax = sparkSession.sessionState.conf.parallelPartitionDiscoveryParallelism)
   }

 }
